### PR TITLE
feat: edit user page

### PR DESCRIPTION
**Pull request by ranerhei**
Made a page to edit users
Made admin page so that it can redirect to edit other users

### CODE REVIEW

File: components/user/EditForm.vue [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+      displayErrorAlert('alert.error.form.invalid')
+      return
+    }
+
+    if (state.userType !== 'company') {
+      state.companyUID = null
+    }
+
+    // update user
+    await useAsyncData('update-user', async () => {
```
Comment (AasmundN) [inline components/user/EditForm.vue:124]: This is bloated as hell, we need to fix it at some point, but not now. See issue #285

File: components/user/EditForm.vue [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+          v-model="state.name"
+          :content="{
+            label: $t('edit.user.attributes.name'),
+          }"
+          :rules="[useRequiredInput]"
+        />
+      </VRow>
+
+      <!-- email -->
+      <VRow>
```
Comment (AasmundN) [inline components/user/EditForm.vue:207]: Should probably verify with a regex that the mail has a valid mail format.

File: composables/useFirebaseAuth.ts [hunk 3]
```diff
@@ -168,3 +168,23 @@ export const requestPasswordReset = (email: string) => {
   const auth = getAuth()
   return sendPasswordResetEmail(auth, email)
 }
+
+// update name of user
+export const updateName = async (newName: string | null | undefined) => {
```
Comment (AasmundN) [inline composables/useFirebaseAuth.ts]: Unused

File: server/api/user/index.put.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  const isAdmin = hasAccess(user, ['admin'])
+
+  if (!isAdmin && uid !== user.uid)
+    throw createError({
+      statusCode: 401,
+      statusMessage: 'User not authorized',
+    })
+  const displayName = newData.name
+  // update name with ID
+  auth.updateUser(uid, { displayName })
```
Comment (AasmundN) [inline server/api/user/index.put.ts:23]: Update mail also

File: composables/useFirebaseAuth.ts (hunk 3 above)
Comment (ranerhei) [reply composables/useFirebaseAuth.ts]: removed

File: server/api/user/index.put.ts (hunk 4 above)
Comment (ranerhei) [reply server/api/user/index.put.ts]: fixed

File: components/user/EditForm.vue (hunk 2 above)
Comment (ranerhei) [reply components/user/EditForm.vue:206]: fixed

**Review by AasmundN — APPROVED** [review]

**Pull request merged**